pkg/provider/azure_vmss_cache.go

```diff
@@ -57,6 +57,11 @@ type vmssEntry struct {
 	lastUpdate    time.Time
 }
 
+type availabilitySetEntry struct {
```

Comment:
And the cache type changes above is required, then I don't think this entry is needed since the cache has already been force refreshed.

Reply:
See my comment above. This is needed in order to store the state of nodeNames at the time the cache entry was created in order to determine if it's worth doing a refresh of the cache. We're not doing a force refresh unless 1) the VM is not found, and 2) the node is new to the cluster